pulser/parametrized/variable.py

```diff
@@ -110,18 +110,18 @@ class _VariableItem(Parametrized, OpSupport):
     key: Union[int, slice]
 
     @property
-    def variables(self):
+    def variables(self) -> Dict:
         return self.var.variables
 
-    def build(self):
+    def build(self) -> Union[List, float, None]:
```

Comment:
Try `Union[ArrayLike, str, float, int]` for the return type. If it doesn't work because `Parametrized` is not type-hinted, just let it go too.